### PR TITLE
CORDA-654 Remaining key constants

### PR DESCRIPTION
**Pull request by andr3ej**
review #2217 first

### CODE REVIEW

File: client/jackson/src/test/kotlin/net/corda/client/jackson/JacksonSupportTest.kt [hunk 1]
```diff
@@ -20,9 +21,12 @@ import java.util.*
...
 
 class JacksonSupportTest {
-    companion object {
-        private val SEED = BigInteger.valueOf(20170922L)
+    private companion object {
+        val SEED = BigInteger.valueOf(20170922L)!!
         val mapper = JacksonSupport.createNonRpcMapper()
+        val ALICE_PUBKEY = TestIdentity(ALICE_NAME, 70).pubkey
+        val DUMMY_NOTARY = TestIdentity(DUMMY_NOTARY_NAME, 20).party
+        val MINI_CORP = TestIdentity(CordaX500Name("MiniCorp", "London", "GB")).party
```
Comment (andr3ej) [inline client/jackson/src/test/kotlin/net/corda/client/jackson/JacksonSupportTest.kt]: i've kept name constants used in demos and inlined those that are not

File: client/rpc/src/integration-test/java/net/corda/client/rpc/CordaRPCJavaClientTest.java [hunk 2]
```diff
@@ -56,7 +56,7 @@ private void login(String username, String password) {
 
     @Before
     public void setUp() throws ExecutionException, InterruptedException {
-        node = startNode(getALICE().getName(), 1, singletonList(rpcUser));
+        node = startNode(getALICE_NAME(), 1, singletonList(rpcUser));
```
Comment (andr3ej) [inline client/rpc/src/integration-test/java/net/corda/client/rpc/CordaRPCJavaClientTest.java]: @\JvmField will be in a subsequent pr

File: core/src/test/kotlin/net/corda/core/flows/CollectSignaturesFlowTests.kt [hunk 3]
```diff
@@ -129,7 +134,7 @@ class CollectSignaturesFlowTests {
     @Test
     fun `fails when not signed by initiator`() {
         val onePartyDummyContract = DummyContract.generateInitial(1337, notary, alice.ref(1))
-        val miniCorpServices = MockServices(listOf("net.corda.testing.contracts"), rigorousMock(), MINI_CORP.name, MINI_CORP_KEY)
+        val miniCorpServices = MockServices(listOf("net.corda.testing.contracts"), rigorousMock(), miniCorp)
```
Comment (andr3ej) [inline core/src/test/kotlin/net/corda/core/flows/CollectSignaturesFlowTests.kt]: MockServices now supports TestIdentity for name and first key

File: docs/source/example-code/src/main/kotlin/net/corda/docs/tutorial/tearoffs/TutorialTearOffs.kt [hunk 4]
```diff
@@ -4,16 +4,16 @@ import net.corda.core.contracts.Command
...
 import net.corda.core.transactions.FilteredTransactionVerificationException
 import net.corda.core.transactions.SignedTransaction
 import net.corda.finance.contracts.Fix
-import net.corda.testing.ALICE
 import java.util.function.Predicate
 
 fun main(args: Array<String>) {
     // Typealias to make the example coherent.
-    val oracle = ALICE
+    val oracle = Any() as AbstractParty
```
Comment (andr3ej) [inline docs/source/example-code/src/main/kotlin/net/corda/docs/tutorial/tearoffs/TutorialTearOffs.kt]: this main doesn't work anyway, see next line

File: experimental/src/test/kotlin/net/corda/finance/contracts/universal/Cap.kt [hunk 5]
```diff
@@ -2,30 +2,27 @@ package net.corda.finance.contracts.universal
...
 import net.corda.testing.node.MockServices
-import net.corda.testing.node.makeTestIdentityService
 import org.junit.Ignore
 import org.junit.Rule
 import org.junit.Test
-import java.math.BigInteger
 import java.time.Instant
 import java.time.LocalDate
 
+internal val DUMMY_NOTARY = TestIdentity(DUMMY_NOTARY_NAME, 20).party
```
Comment (andr3ej) [inline experimental/src/test/kotlin/net/corda/finance/contracts/universal/Cap.kt]: shared among the tests in this package

File: experimental/src/test/kotlin/net/corda/finance/contracts/universal/ContractDefinition.kt [hunk 6]
```diff
@@ -1,20 +1,17 @@
...
 import java.util.*
 import kotlin.test.assertEquals
 import kotlin.test.assertTrue
 
 // Test parties
-val acmeCorp = Party(ALICE.name, generateKeyPair().public)
-val highStreetBank = Party(MEGA_CORP.name, generateKeyPair().public)
-val momAndPop = Party(MINI_CORP.name, generateKeyPair().public)
-
+val acmeCorp = TestIdentity(CordaX500Name("Alice Corp", "Madrid", "ES")).party
```
Comment (andr3ej) [inline experimental/src/test/kotlin/net/corda/finance/contracts/universal/ContractDefinition.kt]: i haven't used the name constant as this isn't really alice - different variable name, no fixed entropy

File: finance/src/test/kotlin/net/corda/finance/contracts/asset/CashTests.kt [hunk 7]
```diff
@@ -30,19 +29,39 @@ import org.junit.After
...
+        val MINI_CORP get() = miniCorp.party
+        val MINI_CORP_IDENTITY get() = miniCorp.identity
+        val MINI_CORP_KEY get() = miniCorp.key
+        val MINI_CORP_PUBKEY get() = miniCorp.pubkey
     }
 
     @Rule
     @JvmField
     val testSerialization = SerializationEnvironmentRule()
-    private val defaultRef = OpaqueBytes(ByteArray(1, { 1 }))
```
Comment (andr3ej) [inline finance/src/test/kotlin/net/corda/finance/contracts/asset/CashTests.kt]: honestly

File: node/src/integration-test/kotlin/net/corda/node/SSHServerTest.kt [hunk 8]
```diff
@@ -105,7 +104,7 @@ class SSHServerTest {
             assertTrue(session.isConnected)
 
             val channel = session.openChannel("exec") as ChannelExec
-            channel.setCommand("start FlowICannotRun otherParty: \"O=Alice Corp,L=Madrid,C=ES\"")
+            channel.setCommand("start FlowICannotRun otherParty: \"${ALICE_NAME}\"")
```
Comment (andr3ej) [inline node/src/integration-test/kotlin/net/corda/node/SSHServerTest.kt]: maybe this test shouldn't use the constant at all, anyway at least it won't break now if someone changes ALICE_NAME assuming they don't add a double quote

File: node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java [hunk 9]
```diff
@@ -36,30 +35,31 @@
...
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class VaultQueryJavaTests {
-    private static final CordaX500Name DUMMY_CASH_ISSUER_NAME = new CordaX500Name("Snake Oil Issuer", "London", "GB");
-    private static final KeyPair DUMMY_CASH_ISSUER_KEY = entropyToKeyPair(BigInteger.valueOf(10));
-    private static final PartyAndCertificate DUMMY_CASH_ISSUER_IDENTITY = getTestPartyAndCertificate(new Party(DUMMY_CASH_ISSUER_NAME, DUMMY_CASH_ISSUER_KEY.getPublic()));
-    private static final PartyAndReference DUMMY_CASH_ISSUER = DUMMY_CASH_ISSUER_IDENTITY.getParty().ref((byte) 1);
+    private static final TestIdentity BOC = new TestIdentity(getBOC_NAME());
+    private static final Party CHARLIE = new TestIdentity(getCHARLIE_NAME(), 90L).getParty();
+    private static final TestIdentity DUMMY_CASH_ISSUER_INFO = new TestIdentity(new CordaX500Name("Snake Oil Issuer", "London", "GB"), 10L);
```
Comment (andr3ej) [inline node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java]: i hate this naming but DUMMY_CASH_ISSUER is already taken. suggestions welcome

File: node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java [hunk 10]
```diff
@@ -448,9 +448,9 @@ public void aggregateFunctionsSumByIssuerAndCurrencyAndSortByAggregateSum() {
             Amount<Currency> pounds300 = new Amount<>(300, Currency.getInstance("GBP"));
             Amount<Currency> pounds400 = new Amount<>(400, Currency.getInstance("GBP"));
             vaultFiller.fillWithSomeTestCash(dollars100, issuerServices, 1, DUMMY_CASH_ISSUER);
-            vaultFiller.fillWithSomeTestCash(dollars200, issuerServices, 2, getBOC().ref(new OpaqueBytes("1".getBytes())));
+            vaultFiller.fillWithSomeTestCash(dollars200, issuerServices, 2, BOC.ref((byte) '1'));
```
Comment (andr3ej) [inline node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java]: that's what it was doing before (assuming an ascii-compatible platform encoding) and i don't feel like fiddling with it

File: samples/bank-of-corda-demo/src/main/kotlin/net/corda/bank/BankOfCordaCordform.kt [hunk 11]
```diff
@@ -34,7 +34,7 @@ class BankOfCordaCordform : CordformDefinition() {
             rpcPort(10003)
         }
         node {
-            name(CordaX500Name(organisation = "BankOfCorda", locality = "London", country = "GB"))
+            name(BOC_NAME)
```
Comment (andr3ej) [inline samples/bank-of-corda-demo/src/main/kotlin/net/corda/bank/BankOfCordaCordform.kt]: presumably this should match BOC_NAME as used below

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt [hunk 12]
```diff
@@ -52,7 +52,7 @@ open class MockServices private constructor(
         override val validatedTransactions: WritableTransactionStorage,
         override val identityService: IdentityServiceInternal,
         private val initialIdentityName: CordaX500Name,
-        vararg val keys: KeyPair
+        val keys: Array<out KeyPair>
```
Comment (andr3ej) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt]: no point it being vararg if it's private

File: testing/test-utils/src/main/kotlin/net/corda/testing/TestDSL.kt [hunk 13]
```diff
@@ -255,22 +254,19 @@ data class TestLedgerDSLInterpreter private constructor(
      */
     private fun fillTransaction(transactionBuilder: TransactionBuilder) {
         if (transactionBuilder.commands().isEmpty()) transactionBuilder.addCommand(dummyCommand())
-        if (transactionBuilder.inputStates().isEmpty() && transactionBuilder.outputStates().isEmpty()) {
-            transactionBuilder.addOutputState(DummyContract.SingleOwnerState(owner = ALICE), DummyContract.PROGRAM_ID)
-        }
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/TestDSL.kt]: fortunately everything that calls unverifiedTransaction adds an input/output state so this never fires

File: testing/test-utils/src/main/kotlin/net/corda/testing/TestDSL.kt [hunk 14]
```diff
@@ -318,40 +314,3 @@ data class TestLedgerDSLInterpreter private constructor(
...
-    SignedTransaction(wtx, signatures)
-}
-
-/**
- * Signs all transactions in the ledger.
- * @param extraKeys extra keys to sign transactions with.
- * @return List of [SignedTransaction]s.
- */
-fun LedgerDSL<TestTransactionDSLInterpreter, TestLedgerDSLInterpreter>.signAll(
-        vararg extraKeys: KeyPair) = signAll(this.interpreter.wireTransactions, extraKeys.toList())
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/TestDSL.kt]: unused stuff

File: testing/test-utils/src/main/kotlin/net/corda/testing/TransactionDSLInterpreter.kt [hunk 15]
```diff
@@ -88,7 +88,7 @@ class TransactionDSL<out T : TransactionDSLInterpreter>(interpreter: T, private
      */
     fun input(contractClassName: ContractClassName, state: ContractState) {
         val transaction = ledgerInterpreter._unverifiedTransaction(null, TransactionBuilder(notary)) {
-            output(contractClassName, null, DUMMY_NOTARY, null, AlwaysAcceptAttachmentConstraint, state)
+            output(contractClassName, null, notary, null, AlwaysAcceptAttachmentConstraint, state)
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/TransactionDSLInterpreter.kt]: oops, missed this one in the other pr

File: docs/source/example-code/src/main/kotlin/net/corda/docs/tutorial/tearoffs/TutorialTearOffs.kt (hunk 4 above)
Comment (mnesbit) [reply docs/source/example-code/src/main/kotlin/net/corda/docs/tutorial/tearoffs/TutorialTearOffs.kt]: Errr this is meant to be an example to people, but the whole thing is gibberish?

Comment (andr3ej) [reply docs/source/example-code/src/main/kotlin/net/corda/docs/tutorial/tearoffs/TutorialTearOffs.kt]: * fixing this example is definitely outside the scope of this pr, e.g. it probably shouldn't be called main
* it looks like this preamble (which isn't included in the docs) only exists to make the snippets below compile

**Comment by andr3ej** [comment]
i agree ALICE.name is nicer, but i don't have anything to put in ALICE other than the name

File: core/src/test/kotlin/net/corda/core/flows/AttachmentTests.kt [hunk 16]
```diff
@@ -97,12 +96,12 @@ class AttachmentTests {
...
         // Make a node that doesn't do sanity checking at load time.
-        val aliceNode = mockNet.createNode(MockNodeParameters(legalName = ALICE.name), nodeFactory = { args ->
+        val aliceNode = mockNet.createNode(MockNodeParameters(legalName = ALICE_NAME), nodeFactory = { args ->
             object : MockNetwork.MockNode(args) {
                 override fun start() = super.start().apply { attachments.checkAttachmentsOnLoad = false }
             }
         })
-        val bobNode = mockNet.createNode(MockNodeParameters(legalName = BOB.name))
+        val bobNode = mockNet.createNode(MockNodeParameters(legalName = BOB_NAME))
         val alice = aliceNode.services.myInfo.identityFromX500Name(ALICE_NAME)
```
Comment (mnesbit) [inline core/src/test/kotlin/net/corda/core/flows/AttachmentTests.kt]: Do we need the identityFromX500Name? Surely this is just the node's own identity?

**Pull request merged**